src/frontend/components/StudentLiveControlButtons/StudentLeaveDiscussionButton/index.spec.tsx

```diff
@@ -0,0 +1,51 @@
+import { fireEvent, render, screen } from '@testing-library/react';
+import React from 'react';
+
+import { PLAYER_ROUTE } from 'components/routes';
+import { wrapInIntlProvider } from 'utils/tests/intl';
+import { wrapInRouter } from 'utils/tests/router';
+import * as mockWindow from 'utils/window';
+
+import { StudentLeaveDiscussionButton } from '.';
+
+jest.mock('utils/window', () => ({
+  converse: {
+    participantLeaves: jest.fn(),
+  },
+}));
+
+describe('<StudentLeaveDiscussionButton />', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('renders the leave button', () => {
+    render(wrapInIntlProvider(wrapInRouter(<StudentLeaveDiscussionButton />)));
+
+    screen.getByRole('button', { name: 'Leave discussion' });
+  });
+
+  it('clicks on the leave button and is redirected to the player', () => {
+    render(
+      wrapInIntlProvider(
+        wrapInRouter(<StudentLeaveDiscussionButton />, [
+          {
+            path: PLAYER_ROUTE(),
+            render: () => {
+              return <span>{'video player'}</span>;
+            },
+          },
+        ]),
+      ),
+    );
+
+    const leaveButton = screen.getByRole('button', {
+      name: 'Leave discussion',
+    });
+
+    fireEvent.click(leaveButton);
+
+    screen.getByText('video player');
+    expect(mockWindow.converse.participantLeaves).toHaveBeenCalled();
```

Comment:
You should also checked that `history.push` has been called with the expected location.

Reply:
I kinda disagree here, we are already checking the redirection is made, I'm not sure we should test how we did it

Reply:
And I agree with @RVassili we never test in our test if `history.push` has been called. This is the purpose of our helper `wrapInRouter` to don't do that

Reply:
Oh yes ! So I'll just precise the route:
```ts
path: PLAYER_ROUTE(modelName.VIDEOS),
```

Reply:
https://github.com/openfun/marsha/pull/1249/commits/db9d11082ce9e5de7d90bc5f56f1c87e12bc9302